modules/sfp_seon.py

```diff
@@ -0,0 +1,263 @@
+        'flags': ["apikey"],
+        'useCases': ["Footprint", "Investigate", "Passive"],
+        'categories': ["Real World"],
+        'dataSource': {
+            'website': "https://seon.io/",
+            'model': "COMMERCIAL_ONLY",
+            'references': [
+                "https://developers.seon.io/"
+            ],
+            'favIcon': "https://seon.io/assets/favicons/favicon-16x16.png",
+            'logo': "https://seon.io/assets/favicons/apple-touch-icon-152.png",
+            'description': "SEON Fraud Prevention tools help organisations reduce "
+            "the costs and resources lost to fraud. Spot fake accounts, slash manual reviews and cut chargebacks now.",
+        }
+    }
+
+    # Default options
+    opts = {
+        'api_key': '',
+        'fraud_threshold': 80,
+    }
+
+    # Option descriptions
+    optdescs = {
+        'api_key': "seon.io API Key",
+        'fraud_threshold': 'Minimum fraud score for target to be marked as malicious (0-100)',
+    }
+
+    results = None
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return [
+            "IP_ADDRESS",
+            "IPV6_ADDRESS",
+            "EMAILADDR",
+            "PHONE_NUMBER"
+        ]
+
+    # What events this module produces
+    # This is to support the end user in selecting modules based on events
+    # produced.
+    def producedEvents(self):
+        return [
+            "GEOINFO",
+            "MALICIOUS_IPADDR",
+            "TCP_PORT_OPEN",
+            "MALICIOUS_EMAILADDR",
+            "EMAILADDR_DELIVERABLE",
+            "EMAILADDR_UNDELIVERABLE",
+            "SOCIAL_MEDIA",
+            "HUMAN_NAME",
+            "COMPANY_NAME",
+            "EMAILADDR_COMPROMISED",
+            "MALICIOUS_PHONE_NUMBER",
+            "PROVIDER_TELCO",
+            "PHONE_NUMBER_TYPE",
+            "WEBSERVER_TECHNOLOGY",
+            "RAW_RIR_DATA"
+        ]
+
+    def query(self, qry, eventName):
+        if eventName == "IP_ADDRESS" or eventName == "IPV6_ADDRESS":
+            queryString = f"https://api.seon.io/SeonRestService/ip-api/v1.0/{qry}"
+        elif eventName == "EMAILADDR":
+            queryString = f"https://api.seon.io/SeonRestService/email-api/v2.0/{qry}"
+        elif eventName == "PHONE_NUMBER":
+            queryString = f"https://api.seon.io/SeonRestService/phone-api/v1.0/{qry}"
+
+        headers = {
+            'Accept': "application/json",
+            'X-API-KEY': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        return json.loads(res['content'])
```

Comment:
Should have the regular 404, 429, etc. checks here before returning and setting `errorState` if appropriate.